components/mosart/src/riverroute/MOSART_physics_mod.F90

```diff
@@ -876,11 +878,7 @@ subroutine Routing_KW(iunit, nt, theDeltaT)
               TRunoff%erout(iunit,nt) = -TRunoff%vr(iunit,nt) * TRunoff%mr(iunit,nt)
               if(-TRunoff%erout(iunit,nt) > TINYVALUE .and. TRunoff%wr(iunit,nt) + &
                  (TRunoff%erlateral(iunit,nt) + TRunoff%erin(iunit,nt) + TRunoff%erout(iunit,nt)) * theDeltaT < TINYVALUE) then
-                 if (sediflag) then
-                  TRunoff%erout(iunit,nt) = -(TRunoff%erlateral(iunit,nt) + TRunoff%erin(iunit,nt) + TRunoff%wr(iunit,nt)*MaxStorageDepleted/ theDeltaT)
-                 else
-                  TRunoff%erout(iunit,nt) = -(TRunoff%erlateral(iunit,nt) + TRunoff%erin(iunit,nt) + TRunoff%wr(iunit,nt)/ theDeltaT)
-                 end if
+                 TRunoff%erout(iunit,nt) = -(TRunoff%erlateral(iunit,nt) + TRunoff%erin(iunit,nt) + TRunoff%wr(iunit,nt)*MaxStorageDepleted/ theDeltaT)
```

Comment:
I think this will result in non BFB in the MOSART simulation.

Reply:
You are right. This is non-BFB. It won't affect the overall model simulation (hence science), but it does change the results slightly for the rare cases when a whole channel storage is completely depleted. I added the "non-BFB" label.

Reply:
Since its potentially non-BFB, this fix would have to go in with 3.1